lib/rescue_from_duplicate/active_record/extension.rb

```diff
@@ -24,22 +24,28 @@ def _rescue_from_duplicate_handlers
 
     def create_or_update(*params, &block)
       super
-    rescue ActiveRecord::RecordNotUnique => exception
-      handler = exception_handler(exception)
+    rescue ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid => exception
```

Comment:
when the logger is involved yes, https://github.com/github/rails/blob/3-2-github/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L289-L292 =(